pygis/docs/e_extraction.md

```diff
@@ -0,0 +1,350 @@
+
+# Convert shapely object to a GeoDataFrame
+poly = gpd.GeoDataFrame(attributes_df, geometry = 'geometry', crs = "EPSG:2768")
+```
+
+We'll define some functions to make displaying and mapping our results a bit easier.
+
+```{code-cell} ipython3
+def display_table(table_name, attribute_table):
+    '''Display the first and last five rows of attribute table.'''
+
+    # Print title
+    print("Attribute Table: {}".format(table_name))
+
+    # Print number of rows and columns
+    print("\nTable shape (rows, columns): {}".format(attribute_table.shape))
+
+    # Display first five rows of attribute table
+    print("\nFirst five rows:")
+    display(attribute_table.head())
+
+    # Display last five rows of attribute table
+    print("\nLast five rows:")
+    display(attribute_table.tail())
+
+
+def plot_df(result_name, result_df, result_geom_type, area = None):
+    '''Plot the result on a map and add the outlines of the original shapefiles.'''
+
+    # Create subplots
+    fig, ax = plt.subplots(1, 1, figsize = (10, 10))
+
+    # Plot data depending on vector type
+    # For points
+    if result_geom_type == "point":
+
+        # Plot data
+        counties.plot(ax = ax, color = 'none', edgecolor = 'dimgray')
+        wells.plot(ax = ax, marker = 'o', color = 'dimgray', markersize = 3)
+        result_df.plot(ax = ax, marker = 'o', color = 'dodgerblue', markersize = 3)
+
+    # For polygons
+    else:
+
+        # Plot overlay data
+        result_df.plot(ax = ax, cmap = 'Set2', edgecolor = 'black')
+
+        # Plot outlines of original shapefiles
+        counties.plot(ax = ax, color = 'none', edgecolor = 'dimgray')
+
+    # Add additional outlined boundary if specified
+    if area is not None:
+
+        # Plot data
+        area.plot(ax = ax, color = 'none', edgecolor = 'lightseagreen', linewidth = 3)
+
+    # Else, pass
+    else:
+        pass
+
+    # Stylize plots
+    plt.style.use('bmh')
+
+    # Set title
+    ax.set_title(result_name, fontdict = {'fontsize': '15', 'fontweight' : '3'})
+```
+
+Let's take a look at what our input data looks like.
+
+```{code-cell} ipython3
+# Create subplots
+fig, ax = plt.subplots(1, 1, figsize = (10, 10))
+
+# Plot data
+counties.plot(ax = ax, color = 'bisque', edgecolor = 'dimgray')
+wells.plot(ax = ax, marker = 'o', color = 'dodgerblue', markersize = 3)
+poly.plot(ax = ax, color = 'aquamarine', edgecolor = 'lightseagreen', alpha = 0.55)
+
+# Stylize plots
+plt.style.use('bmh')
+
+# Set title
+ax.set_title('San Francisco Bay Area', fontdict = {'fontsize': '15', 'fontweight' : '3'})
+```
+
+## Clip
+
+Clip extracts and keeps only the geometries of a vector feature that are within extent of another vector feature (think of it like a cookie-cutter or mask). We can use  `clip()` in `geopandas`, with the first parameter being the vector that will be clipped and the second parameter being the vector that will define the extent of the clip. All attributes for the resulting clipped vector will be kept.
```

Comment:
I like the 'cookie cutter', any analogy is a good one

Reply:
Italize this *"All attributes for the resulting clipped vector will be kept."* its important